ament_index_python/test/test_ament_index_python.py

```diff
@@ -144,26 +133,15 @@ def test_get_package_prefix():
     set_ament_prefix_path(['prefix1', 'prefix2'])
 
     def get_package_prefix_basename(package_name):
-        return os.path.basename(get_package_prefix(package_name))
+        return PurePath(get_package_prefix(package_name)).name
 
     assert get_package_prefix_basename('foo') == 'prefix1', "Expected 'foo' in 'prefix1'"
     # found in both prefix1 and prefix2, but prefix1 is ahead on the APP
     assert get_package_prefix_basename('bar') == 'prefix1', "Expected 'bar' in 'prefix2'"
     assert get_package_prefix_basename('baz') == 'prefix2', "Expected 'baz' in 'prefix2'"
 
-    try:
-        get_package_prefix('does_not_exist')
-    except PackageNotFoundError:
-        pass
-    except Exception as exc:
-        assert False, 'Expected PackageNotFoundError, got: {}'.format(type(exc))
-
-    try:
+    with pytest.raises(PackageNotFoundError):
         get_package_prefix('does_not_exist')
-    except KeyError:
-        pass
-    except Exception as exc:
-        assert False, 'Expected KeyError or subclass, got: {}'.format(type(exc))
```

Comment:
Should the second test - expecting a `KeyError` - be kept? Otherwise that base class might be removed from `PackageNotFoundError` breaking code relying on that.